### ISSUE TITLE
Cannot GET /rest/oauth2-credential/callback

### ISSUE THREAD

**Issue by javierfuentesm** [n8n-io/n8n#10973]
### Bug Description

![image](https://github.com/user-attachments/assets/04d8730f-4880-482e-869b-8dfec2017e56)

### To Reproduce

Add a google drive account

![image](https://github.com/user-attachments/assets/fe5646cd-e718-42bd-8e3b-ca7dac88f48f)

When you click Sign in with google, it goes through the normal flow but at the end
![Opera 2024-09-25 8 23 04 p m](https://github.com/user-attachments/assets/fa5386d7-532d-47c1-aba4-d797e2962fe4)
![Opera 2024-09-25 8 27 14 p m](https://github.com/user-attachments/assets/7af1c79c-b6a0-47e4-8c00-9e3671da2c06)
![Opera 2024-09-25 8 27 24 p m](https://github.com/user-attachments/assets/02dacc8f-3212-4b12-ba9d-f2eda5a4c2d2)

I'm using railway:
![image](https://github.com/user-attachments/assets/9c2f4476-39aa-4c53-a331-f6bd771561d5)

### Expected behavior

To get authenticated

### Operating System

linux

### n8n Version

 1.60.1

### Node.js Version

im not sure

### Database

PostgreSQL

### Execution mode

main (default)

There is something weird in the logs some times sometimes; the endpoint returns 204 (but it never passes in the FE)
![Opera 2024-09-25 8 48 50 p m](https://github.com/user-attachments/assets/199ede3f-6819-4ecc-8661-a1813711189e)

**Comment by Joffcom** [comment]
Hey @javierfuentesm,

We have created an internal ticket to look into this which we will be tracking as "GHC-256"

**Comment by javierfuentesm** [comment]
> Hey @javierfuentesm
> 
> This looks like it is a configuration issue, for oauth the user needs to be sent to the primary instance not a webhook worker.
> 
> Can you set `N8N_EDITOR_BASE_URL` to the url for the primary instance then restart and check that the oauth redirect url has changed and try again.
> 
> Also make sure that your load balancer is configured to send `/webhook-test/` URLs to your main instance and `/webhook/` to your webhook worker instance.

Thank you! i didn't know that 
![Opera 2024-09-25 11 39 57 p m](https://github.com/user-attachments/assets/75f964f0-b1e5-4655-9eaa-5526744653a0)
but can you guide me here?  how do I do this? : "Also make sure that your load balancer is configured to send /webhook-test/ URLs to your main instance and /webhook/ to your webhook worker instance."

**Comment by javierfuentesm** [comment]
should i get a different env variable ? for the chat trigger? , i get this 
![image](https://github.com/user-attachments/assets/4a67bdb2-e976-4304-8c52-81748249c289)
and when i open that URL 
![image](https://github.com/user-attachments/assets/75d7d36e-fe7d-4ed7-a523-5453c62eae72)
I get this error, is trying to connect to localhost, is a another env variable missing?

**Comment by Joffcom** [comment]
Hey @javierfuentesm 

We don't have specific instructions on configuring a load balancer to make sure the traffic goes to the correct location as it depends on what you are using so we kind of expect you to know how to do this already.

With the chat trigger assuming you have `WEBHOOK_URL` set as well it should work but I will need to test that to make sure we have not done anything odd.

Out of interest did you set up n8n following our documentation or did you use an existing render template that someone else made?

**Comment by Joffcom** [comment]
Hey @javierfuentesm 

We don't have specific instructions on configuring a load balancer to make sure the traffic goes to the correct location as it depends on what you are using so we kind of expect you to know how to do this already.

With the chat trigger assuming you have `WEBHOOK_URL` set as well it should work but I will need to test that to make sure we have not done anything odd.

Out of interest did you set up n8n following our documentation or did you use an existing render template that someone else made?

**Comment by javierfuentesm** [comment]
> Hey @javierfuentesm
> 
> We don't have specific instructions on configuring a load balancer to make sure the traffic goes to the correct location as it depends on what you are using so we kind of expect you to know how to do this already.
> 
> With the chat trigger assuming you have `WEBHOOK_URL` set as well it should work but I will need to test that to make sure we have not done anything odd.
> 
> Out of interest did you set up n8n following our documentation or did you use an existing render template that someone else made?

I used the railway template for n8n , and apparently, it was wrong , I'm not even sure i have a load balancer
![image](https://github.com/user-attachments/assets/c5219b67-11d9-4e03-af7c-afcd853d8f00)

and yes, I have that URL, and the UI on the bot opens correctly, but in the network requests, I see that it is failing, and it is trying to call the local. Do you have any idea why?
![image](https://github.com/user-attachments/assets/52403fc5-85c3-405c-b874-b7ec1cea2266)

**Comment by javierfuentesm** [comment]
ok it look like i had to add that env variable to the webhook processor too, and now it is responding but any idea why it looks like this ?
![image](https://github.com/user-attachments/assets/ac7b8a2f-af14-425b-bb64-0a67951f77ba)

**Comment by Joffcom** [comment]
Hey @javierfuentesm

that issue looks like a known issue with the chat trigger when using certain AI Agents, This has been reported as https://github.com/n8n-io/n8n/issues/10446 and is not yet fixed. 

I think for now we can close this issue and when the response is resolved we will update the other issue.

**Comment by rhynoverde** [comment]
I tried yours our from the list in railway- was that the wrong way to do it. I couldn't get this oauth to work so I switched to a simpler n8n with postgres only.

Wondering at what level of traffic something like yours would be necessary. My application will only send a few webhooks an hour to start between 0 and maybe 20 or 30 to start. 

But over time that will grow so interested in learning how something like this helps scale.

But I could not get the oath to work and I was getting the processor url in the callback in n8n connection which gave the error that led me here. But switching the url to the primary-production one did not help either. Don't know if there is some variable I would need to change or what I would need to do to make it connect.
